src/nvim/lua/converter.c

```diff
@@ -1299,3 +1299,25 @@ void nlua_init_types(lua_State *const lstate)
 
   lua_rawset(lstate, -3);
 }
+
+
+void nlua_pop_keydict(lua_State *L, void *retval, field_hash hashy, Error *err)
+{
+  lua_pushnil(L);  // [dict, nil]
+  while (lua_next(L, -2)) {
+    // [dict, key, value]
+    size_t len;
+    const char *s = lua_tolstring(L, -2, &len);
+    Object *field = hashy(retval, s, len);
+    if (!field) {
+      api_set_error(err, kErrorTypeValidation, "unexpected key: %.*s", (int)len, s);
```

Comment:
"invalid key" is used in most of the API I think
```suggestion
      api_set_error(err, kErrorTypeValidation, "invalid key: %.*s", (int)len, s);
```

Reply:
not as far as the code is telling me (zero hits on "invalid key")

Reply:
ok it is inconsistent between the interfaces, changing this one for now